packages/admin/admin/src/mui/menu/ItemRouterLink.tsx

```diff
@@ -1,17 +1,52 @@
+import { Badge } from "@mui/material";
 import { ListItemProps } from "@mui/material/ListItem";
 import * as React from "react";
 import { Link, LinkProps, Route } from "react-router-dom";
 
+import { ThemedComponentBaseProps } from "../../helpers/ThemedComponentBaseProps";
 import { MenuItem, MenuItemProps } from "./Item";
 
-interface MenuItemRouterLinkStandardProps {
+interface MenuItemRouterLinkStandardProps
+    extends ThemedComponentBaseProps<{
+        badge: typeof Badge;
+    }> {
     to: string;
+    badgeContent?: React.ReactNode;
 }
 
 export type MenuItemRouterLinkProps = MenuItemRouterLinkStandardProps & MenuItemProps & ListItemProps & LinkProps;
 
-export const MenuItemRouterLink: React.FC<MenuItemRouterLinkProps> = (props) => (
-    <Route path={props.to} strict={false}>
-        {({ match }) => <MenuItem selected={!!match} component={Link} {...props} />}
-    </Route>
-);
+export const MenuItemRouterLink: React.FC<MenuItemRouterLinkProps> = ({ badgeContent, secondaryAction, slotProps, ...restProps }) => {
+    const tempSlotProps = { ...slotProps };
+    const badge = tempSlotProps?.badge;
+    delete tempSlotProps?.badge;
+
+    const computedSecondaryAction = badgeContent ? (
+        <Badge
+            variant={restProps.isMenuOpen ? "standard" : "dot"}
+            color="error"
+            overlap="circular"
+            {...badge}
+            badgeContent={badgeContent}
+            sx={{ marginLeft: 2, ...badge?.sx }}
+        />
+    ) : (
+        secondaryAction
+    );
```

Comment:
nit: Same thing here

```suggestion
    const secondaryAction = badgeContent ? (
        <Badge
            variant={restProps.isMenuOpen ? "standard" : "dot"}
            color="error"
            overlap="circular"
            {...badge}
            badgeContent={badgeContent}
            sx={{ marginLeft: 2, ...badge?.sx }}
        />
    ) : (
        passedSecondaryAction
    );
```

Reply:
https://github.com/vivid-planet/comet/pull/2431/commits/50272840f64509e7bc017b1d3bf0052d613c3a36